File: signal-matrix/python/signals/dlp_reachable.py
"""REACHABLE: DLP — called from entrypoint."""
import logging
logger = logging.getLogger(__name__)

def process_patient_record(patient: dict) -> dict:
    """DLP REACHABLE: SSN + DOB logged to stdout."""
    ssn = patient.get("ssn")
    dob = patient.get("dob")
    logger.info(f"Processing patient ssn={ssn} dob={dob}")  # DLP: PII → log
    return {"status": "processed", "id": 42}

File: signal-matrix/javascript/signals/dlp_reachable.js
'use strict';
// REACHABLE: DLP — called from server.js
const axios = require('axios');

function processUserPii(userData) {
    const { ssn, creditCard, email } = userData;
    // DLP REACHABLE: PII logged
    console.log(`Processing: ssn=${ssn} card=${creditCard} email=${email}`);
    // DLP REACHABLE: PII sent to external API
    axios.post('https://analytics.example.com/track', { ssn, creditCard, email });
    return { processed: true };
}

module.exports = { processUserPii };
